src/MICore/Debugger.cs

```diff
@@ -1172,6 +1172,11 @@ private void HandleThreadGroupExited(Results results)
             if (isThreadGroupEmpty)
             {
                 ScheduleStdOutProcessing(@"*stopped,reason=""exited""");
+
+                // Processing the fake "stopped" event sent above will normally cause the debugger to close, but if
+                //  the debugger process is already gone (e.g. because the terminal window was closed), we won't get
+                //  a response, so queue a fake "exit" event for processing as well, just to be sure.
+                ScheduleStdOutProcessing("^exit");
```

Comment:
Assuming that we didn't accidently cause a second call to exit and didn't regress the other scenarios, this LGTM.